### ISSUE TITLE
[java] AvoidUsingOctalValues FN with underscores

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#3400]
**Affects PMD Version:** 6.36.0

**Rule:** [AvoidUsingOctalValues](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#avoidusingoctalvalues)

**Description:** The rule should report octal literals that contain underscores

**Code Sample demonstrating the issue:**

```java
class Test {
  int i = 0_711;
}
```

**Expected outcome:** a violation

PMD should report a violation at line ..., but doesn't. This is a false-negative.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by jsotuyod** [comment]
This is fixed in PMD 7.0.0-rc1. It won't be backported to PMD 6.
